packages/utils/telemetry-utils/src/telemetryEventBatcher.ts

```diff
@@ -14,21 +14,23 @@ import type {
 
 /**
  * Expected type of the custom data passed into the logger.
+ * @internal
  */
-interface IMeasuredCodeResult<TKey extends string> {
+export interface IMeasuredCodeResult<TKey extends string> {
 	/**
 	 * Optional properties to log custom data. The set of properties must be the same for all calls to the `measure` function.
 	 */
 	telemetryProperties?: { readonly [key in TKey]: number };
 }
 
 /**
- * Telemetry class that measures the execution time of a given piece of code and accumulates user defined telemetry metrics ({@link ICustomDataMap}), to finally log an event through the {@link TelemetryEventBatcher.logger | logger} provided to this class when the number of calls to the {@link TelemetryEventBatcher.measure | measure} function reaches the specified by {@link TelemetryEventBatcher.threshold | threshold}.
+ * Telemetry class that measures the execution time of a given piece of code and accumulates user defined telemetry metrics, to finally log an event through the {@link TelemetryEventBatcher.logger | logger} provided to this class when the number of calls to the {@link TelemetryEventBatcher.measure | measure} function reaches the specified by {@link TelemetryEventBatcher.threshold | threshold}.
  *
  * @remarks It is expected to be used for a single event type. If the set of `telemetryProperties` is different for different events, a separate `TelemetryEventBatcher` should be created for each event type.
  * @typeparam TMetrics - The set of keys that should be logged.
  * E.g., `keyof Foo` for logging properties `bar` and `baz` from `type Foo = { bar: number, baz: number }`.
  *
+ * @sealed
```

Comment:
Marked as sealed by default to forbid subclassing. Can be removed later as needed.